### PR TITLE
Strawberry guid robustness

### LINKED ISSUE #3562: Guid serialization in strawberry shake only accepts one format.
StrawberryShake only seems to recognize uuid values in "D" form (aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee). Is there a way to get it to use "N" form (aaaaaaaabbbbccccddddeeeeeeeeeeee)?

### PR DESCRIPTION
**Pull request by dmg-hamann**
set guid to string serialization

- enhanced UuidSerializer to use Guid.Parse

Closes #3562

### CODE REVIEW

**Comment by michaelstaib** [comment]
Already skimmed through it .... look really good. Thanks for putting the extra effort into this.... will do a more thorough review tomorrow.

**Review by PascalSenn — CHANGES_REQUESTED** [review]
Found this one as i skimmed through. Looks very good 👍

File: src/StrawberryShake/Client/src/Core/Serialization/UuidSerializer.cs [hunk 1]
```diff
@@ -5,11 +5,25 @@ namespace StrawberryShake.Serialization
...
+            if (Guid.TryParse(serializedValue, out Guid guid))
+            {
+                return guid;
+            }
+            throw ThrowHelper.UuidSerializer_CouldNotParse(serializedValue);
+        }
+
+        protected override string Format(Guid runtimeValue)
+        {
+            return runtimeValue.ToString();
```
Comment (michaelstaib) [inline src/StrawberryShake/Client/src/Core/Serialization/UuidSerializer.cs:26]: We should use a specific format here. We could have a optional format in the constructor.

**Comment by michaelstaib** [comment]
Looks great

**Comment by michaelstaib** [comment]
I think we can merge this one tomorrow.

File: src/StrawberryShake/Client/src/Core/Serialization/UuidSerializer.cs (hunk 1 above)
Comment (dmg-hamann) [reply src/StrawberryShake/Client/src/Core/Serialization/UuidSerializer.cs]: I updated the UuidSerializer to support optionally other formats.

**Review by michaelstaib — APPROVED** [review]

**Review by PascalSenn — APPROVED** [review]

**Comment by michaelstaib** [comment]
@dmg-hamann great work! Thanks for your help. Once build goes green we will merge.

**Comment by dmg-hamann** [comment]
Awesome. Thanks to you guys for this awesome project :)

**Comment by michaelstaib** [comment]
If you need the feature in 11.x you can do another pr where you cherrypick this change from develop to main. If not it will be included into the next 12 preview. Thanks again from the help and welcome as a contributor.

**Comment by gmiserez** [comment]
Hi, 
Just wanted to thank @dmg-hamann for the fix, as I am currently experiencing this issue on 11.3.0 after migrating from 11.0.0-preview.137.

@dmg-hamann, would you be willing to merge this to the master branch as well? I believe many people who use the release versions of StrawberryShake and HotChocolate (ie 11.X) and use Guids in their Apis might currently be experiencing this issue, so you'll probably make a lot of people happy (definitely me!).

**Comment by dmg-hamann** [comment]
@gmiserez Yeah I will cherry pick the changes and setup a pr

**Pull request merged**